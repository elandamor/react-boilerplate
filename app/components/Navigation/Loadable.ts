/**
 *
 * Asynchronously loads the component for Navigation
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
