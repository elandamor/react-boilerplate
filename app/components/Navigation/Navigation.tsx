import React, { SFC } from 'react';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name Navigation component
 * @description Main navigation for an app.
 * @example
 * <Navigation
 *    links={[
 *      { exact: true, label: 'Home', href: '/' },
 *      { label: 'About', href: '/about' }
 *      { label: 'Contact', href: '/contact' },
 *    ]}
 * />
 */

type Link = {
  className?: string,
  exact?: boolean,
  href: string,
  label: string,
};

interface Props {
  links: object[];
};

const Navigation:SFC<Props> = ({ links }) => (
  <Wrapper>
    <ul>
      {
        links.map((link: Link, index) => (
          <li key={index}>
            <NavLink
              exact={link.exact}
              activeClassName="-active"
              className={classNames('a-nav-item', link.className)}
              to={link.href}
            >
              {link.label}
            </NavLink>
          </li>
        ))
      }
    </ul>
  </Wrapper>
);

export default Navigation;
