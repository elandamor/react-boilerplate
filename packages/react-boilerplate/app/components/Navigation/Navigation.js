import React from 'react';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Wrapper from './styles';

/**
 * @render react
 * @name Navigation component
 * @description Main navigation for an app.
 * @example
 * <Navigation
 *    links={[
 *      { label: 'My List', href: '/my-list' },
 *      { label: 'Top Picks', href: '/top' },
 *      { label: 'Recent', href: '/recent' }
 *    ]}
 * />
 */
/* eslint-disable react/no-array-index-key */
const Navigation = ({ links }) => (
  <Wrapper>
    <ul>
      {
        links.map((link, index) => (
          <li key={index}>
            <NavLink
              exact={link.exact}
              activeClassName="-active"
              className={classNames('a-nav-item', link.className)}
              to={link.href}
              aria-label={link.label}
            >
              {link.label}
            </NavLink>
          </li>
        ))
      }
    </ul>
  </Wrapper>
);

Navigation.propTypes = {
  links: PropTypes.array.isRequired,
};

export default Navigation;
