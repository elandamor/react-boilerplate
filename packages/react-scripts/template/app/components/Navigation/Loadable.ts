/**
 *
 * Asynchronously loads the component for Navigation
 *
 */

import React from 'react';

// @ts-ignore
export default React.lazy(() => import('./index'));
