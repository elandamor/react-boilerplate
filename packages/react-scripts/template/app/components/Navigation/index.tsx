import React, { SFC } from 'react';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name Navigation component
 * @description Main navigation for an app.
 * @example
 * <Navigation
 *    links={[
 *      { exact: true, label: 'Home', href: '/' },
 *      { label: 'About', href: '/about' }
 *      { label: 'Contact', href: '/contact' },
 *    ]}
 * />
 */

// tslint:disable-next-line:interface-over-type-literal
type Link = {
  className?: string,
  exact?: boolean,
  href: string,
  label: string,
};

interface IProps {
  className?: string;
  links: Link[];
}

const Navigation: SFC<IProps> = ({ className, links }) => (
  <Wrapper className={classNames('c-nav', className)}>
    <ul>
      {
        links.map((link: Link, index: number) => (
          <li key={index}>
            <NavLink
              exact={link.exact}
              activeClassName="-active"
              className={classNames('a-nav-item', link.className)}
              to={link.href}
            >
              {link.label}
            </NavLink>
          </li>
        ))
      }
    </ul>
  </Wrapper>
);

export default Navigation;
