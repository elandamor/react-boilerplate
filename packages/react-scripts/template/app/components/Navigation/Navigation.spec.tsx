// Navigation.spec.tsx
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Navigation from './index';

describe('<Navigation />', () => {
  it('should display a navigation component', () => {
    const component = renderer
      .create(
        <Router>
          <Navigation
            links={[
              { exact: true, href: '/', label: 'Home' },
              { href: '/about', label: 'About Us' },
            ]}
          />
        </Router>,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
