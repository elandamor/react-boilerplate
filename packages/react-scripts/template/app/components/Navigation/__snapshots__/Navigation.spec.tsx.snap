// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should display a navigation component 1`] = `
<nav
  className="c-nav "
>
  <ul>
    <li>
      <a
        aria-current="page"
        className="a-nav-item -active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className="a-nav-item"
        href="/about"
        onClick={[Function]}
      >
        About Us
      </a>
    </li>
  </ul>
</nav>
`;
